logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/GetRecentMessagesUseCase.kt

```diff
@@ -3,11 +3,15 @@ package com.wire.kalium.logic.feature.message
 import com.wire.kalium.logic.data.id.ConversationId
 import com.wire.kalium.logic.data.message.Message
 import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.persistence.dao.message.MessageEntity
 import kotlinx.coroutines.flow.Flow
 
 class GetRecentMessagesUseCase(private val messageRepository: MessageRepository) {
 
-    suspend operator fun invoke(conversationId: ConversationId, limit: Int = 100, offset: Int = 0): Flow<List<Message>> {
-        return messageRepository.getMessagesForConversation(conversationId, limit, offset)
-    }
+    suspend operator fun invoke(
+        conversationId: ConversationId,
+        limit: Int = 100,
+        offset: Int = 0,
+        visibility: List<Message.Visibility> = Message.Visibility.values().toList()
```

Comment:
When we call this useCase, which visibilities are requested? because we need the deleted/hidden cases to show deleted/hidden message indicator.